### PR TITLE
Move RecordEncoder documentation

### PR DESCRIPTION
**Pull request by toonn**
The documentation for how to use the Divisible instance to create a
RecordEncoder ended up in the "Miscellaneous" section. This made it very
hard to find, after some searching I ended up in the repository and
found out the documentation *was* included on the hackage page I was
trying to find it on.

I moved it to document the definition of the RecordEncoder newtype to
bring it in line with the RecordDecoder documentation.

(Note: The diff is slightly misleading but moving the (>*<) definition up is the same thing as moving the haddock comment down.)

### CODE REVIEW

File: dhall/src/Dhall.hs [hunk 1]
```diff
@@ -2527,13 +2533,6 @@ injectProject =
 
 -}
 
--- | Infix 'divided'
-(>*<) :: Divisible f => f a -> f b -> f (a, b)
-(>*<) = divided
-
-infixr 5 >*<
-
--- | Intermediate type used for building a `ToDhall` instance for a record
```
Comment (german1608) [inline dhall/src/Dhall.hs:2536]: Why did you remove this line of doc?

Comment (german1608) [reply dhall/src/Dhall.hs:2536]: Oh sorry, I understand what you did. I didn't notice that above `(>*<)` there was the doc for `RecordEncoder`.

**Comment by toonn** [comment]
Yeah, the diff isn't obvious. I *did* remove that one line of documentation because it didn't seem to add much of value and the RecordDecoder docs don't mention the equivalent.
